homeassistant/components/light/tikteck.py

```diff
@@ -0,0 +1,146 @@
+"""
+Support for Tikteck lights.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.tikteck/
+"""
+import logging
+import time
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_DEVICES, CONF_NAME, CONF_PASSWORD
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_RGB_COLOR, SUPPORT_BRIGHTNESS, SUPPORT_RGB_COLOR,
+    Light, PLATFORM_SCHEMA)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['tikteck==0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'tikteck'
+
+SUPPORT_TIKTECK_LED = (SUPPORT_BRIGHTNESS | SUPPORT_RGB_COLOR)
+
+DEVICE_SCHEMA = vol.Schema({
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_DEVICES, default={}): {cv.string: DEVICE_SCHEMA},
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Tikteck platform."""
+    lights = []
+    for address, device_config in config[CONF_DEVICES].items():
+        device = {}
+        device['name'] = device_config[CONF_NAME]
+        device['password'] = device_config[CONF_PASSWORD]
+        device['address'] = address
+        light = TikteckLight(device)
+        if light.is_valid:
+            lights.append(light)
+
+    add_devices(lights)
+
+
+class TikteckLight(Light):
+    """Representation of a Tikteck light."""
+
+    def __init__(self, device):
+        """Initialize the light."""
+        import tikteck
+
+        self._name = device['name']
+        self._address = device['address']
+        self._password = device['password']
+        self._brightness = 255
+        self._rgb = [255, 255, 255]
+        self._state = False
+        self.is_valid = True
+        try:
+            self._bulb = tikteck.tikteck(self._address, "Smart Light",
+                                         self._password)
+            self._bulb.connect()
+        except Exception as e:
```

Comment:
You are correct that we don't want to get caught dealing with the implementation details of the lib. The cleanest approach would be to have the lib wrap the exception with it's own error type.